pkg/rbac/sharedData.go

```diff
@@ -210,3 +221,133 @@ func (shared *SharedData) GetManagedClusters(cache *Cache, ctx context.Context)
 	return shared.mcErr
 
 }
+
+func (cache *Cache) GetDisabledClusters(ctx context.Context) (*map[string]struct{}, error) {
+	userData, userDataErr := cache.GetUserData(ctx)
+	if userDataErr != nil {
+		return nil, userDataErr
+	}
+
+	if cache.SharedCacheDisabledClustersValid() {
+		klog.V(5).Info("Search Addon DisabledClusters Cache valid")
+	} else {
+		klog.V(5).Info("DisabledClusters Cache not valid") // - running query to get search addon disabled clusters")
```

Comment:
This is simpler and reduces log verbosity (no log means we used cached data).
```suggestion
	if !cache.SharedCacheDisabledClustersValid() {
		klog.V(5).Info("DisabledClusters cache empty or expired. Querying database.")
```

Reply:
Done, thanks.